src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorDesignTimeNodeWriter.cs

```diff
@@ -343,40 +343,111 @@ public override void WriteComponent(CodeRenderingContext context, ComponentExten
                 throw new ArgumentNullException(nameof(node));
             }
 
-            foreach (var typeArgument in node.TypeArguments)
+            if (node.TypeInferenceNode == null)
```

Comment:
Most of the work here is in the code generator. Basically creating a component has two code paths. I tried to make as much of it reused as is feasible.